frontend/src/my-books/ShelfModal.tsx

```diff
@@ -0,0 +1,99 @@
+
+This program is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+PURPOSE.  See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with this program.
+If not, see <https://www.gnu.org/licenses/>.
+*/
+
+import React, { Component } from "react";
+import Modal, { IModalProps } from "../shared/components/Modal";
+import Button from "@material-ui/core/Button";
+import TextField from "@material-ui/core/TextField";
+import "./ShelfModal.css";
+import Hidden from "@material-ui/core/Hidden";
+
+type MyState = { name: string };
+export default class ShelfModal extends Component<IModalProps, MyState> {
+  constructor(props: never) {
+    super(props);
+    this.state = { name: "" };
+    this.submitShelf = this.submitShelf.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+  }
+
+  handleChange = (event: any) => {
+    this.setState({ name: event.target.value });
+  };
+
+  submitShelf = (event: any) => {
+    alert(`Shelf added: ${this.state.name}`);
+    event.preventDefault();
+  };
+
+  render(): JSX.Element {
+    return (
+      <div>
+        <Modal open={this.props.open} onClose={this.props.onClose}>
+          <div className="shelf-modal-container">
+            <div className="modal-content">
+              <div className="modal-title">Add shelf</div>
+              <div className="shelf-modal-desc-container">
+                <Hidden smDown implementation="css">
+                  <p>Shelf name</p>
+                  <TextField
+                    className="shelfInput"
+                    size="small"
+                    id="outlined-basic"
+                    variant="outlined"
+                    value={this.state.name}
+                    onChange={this.handleChange}
+                  />
+                </Hidden>
+                <Hidden mdUp implementation="css">
+                  <TextField
+                    className="shelfInput"
+                    size="small"
+                    id="name"
+                    variant="outlined"
+                    label="shelf name"
+                    value={this.state.name}
+                    onChange={this.handleChange}
+                  />
+                </Hidden>
+              </div>
+            </div>
+            <div className="modal-form-spacer" />
+
+            <div className="shelf-button-container">
+              <Button
+                className="shelf-modal-button"
+                variant="contained"
+                onClick={this.props.onClose}
+              >
+                Cancel
+              </Button>
+              <Button
+                className="shelf-modal-button"
+                variant="contained"
+                onClick={this.submitShelf}
+                color="primary"
+              >
+                Add shelf
+              </Button>
```

Comment:
You're buttons are displaying correctly for desktop, when they are flexed in a row. However, for mobile, the Cancel button should be below Add Shelf button. In media query, you can use css flex styles to reverse the order of the children. Additionally, please add space between the buttons.

Reply:
In the last commit, I already added 24px to push the button apart in the mobile view, do you mean I should increase the space to about 30px?

Reply:
Can you update your screenshot so we can see how the 24 px looks? Thank you :)